var/spack/repos/builtin/packages/intel-tbb/package.py

```diff
@@ -68,9 +68,12 @@ class IntelTbb(Package):
     depends_on('cmake@3.0.0:', type='build', when='@2017.0:')
 
     # Deactivate use of RTM with GCC when on an OS with an elderly assembler.
-    patch("tbb_gcc_rtm_key.patch", level=0, when='%gcc@4.8.0: os=rhel6')
-    patch("tbb_gcc_rtm_key.patch", level=0, when='%gcc@4.8.0: os=scientific6')
-    patch("tbb_gcc_rtm_key.patch", level=0, when='%gcc@4.8.0: os=centos6')
+    patch("tbb_gcc_rtm_key_2019U1.patch", level=0, when='@2019.1: %gcc@4.8.0: os=rhel6')
```

Comment:
Just to be clear, I think `when='%gcc@4.8.0: os=rhel6'` does not equate with the comment "when on an OS with an elderly assembler."

`gcc@4.8.0:` means every gcc from 4.8.0 up to gcc 9 right?

If so, this still needs to be fixed in this PR, I just wasn't sure of the right way to do it.  (Mainly, what is meant by "elderly")